### ISSUE TITLE
Update ESMF to v8.2.0 

### ISSUE THREAD

**Issue by DusanJovic-NOAA** [ufs-community/ufs-weather-model#893]
UFS Weather Model should be updated to use recently released ESMF version 8.2.0. This version has already been installed on all supported platforms.

I ran the full regression test on Orion and found that the following three tests failed:
```
Test 077 hafs_regional_atm_ocn FAIL
Test 079 hafs_regional_atm_ocn_wav FAIL
Test 086 hafs_regional_datm_cdeps FAIL
```
with the following run time error:
```
20211102 143940.009 INFO            PET305 (InitializeP0): done 
20211102 143944.246 ERROR           PET305 read_impexp_config_mod.F90:189 Not valid - Malformed ocn_export_fields item FORMAT='HYCOM_NAME' 'STANDARD_NAME' 'UNITS' in file: nems.configure 
20211102 143944.246 ERROR           PET305 HYCOM_OceanComp.F90:746 Not valid - Passing error in return code 
20211102 143944.246 ERROR           PET305 EARTH Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:2577 Not valid - Phase 'IPDv00p1' Initialize for modelComp 3: OCN did not return ESMF_SUCCESS 
20211102 143944.246 ERROR           PET305 EARTH Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:1262 Not valid - Passing error in return code 
20211102 143944.246 ERROR           PET305 EARTH Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:457 Not valid - Passing error in return code 
20211102 143944.246 ERROR           PET305 MAIN_NEMS.F90:429 Not valid - Aborting NEMS 
20211102 143944.246 INFO            PET305 Finalizing ESMF
```

**Comment by junwang-noaa** [comment]
@rsdunlapiv @BinLiu-NOAA We have the HAFS tests failed with ESMF 820, has it been tested with HYCOM model? Is there anything we need to change?

**Comment by uturuncoglu** [comment]
@junwang-noaa I think there was a issue related with the nems.configuration related with HYCOM. I am not sure this is the same issue or not but @danrosen25 might have idea about it.

**Comment by rsdunlapiv** [comment]
@junwang-noaa please see https://github.com/ufs-community/ufs-weather-model/issues/824

You need to merge in:
https://github.com/NOAA-EMC/HYCOM-src/pull/1

**Comment by junwang-noaa** [comment]
@uturuncoglu @rsdunlapiv Thanks, this is helpful. Maybe we can ask @danrosen25 to commit the HYCOM PR#824.

**Comment by DeniseWorthen** [comment]
Close; Update was merged in PR #886
